x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherServiceTests.java

```diff
@@ -204,6 +208,36 @@ void stopExecutor() {
         assertThat(watches, hasSize(activeWatchCount));
     }
 
+    public void testPausingWatcherServiceAlsoPausesTriggerService() {
+        String engineType = "foo";
+        TriggerEngine triggerEngine = mock(TriggerEngine.class);
+        when(triggerEngine.type()).thenReturn(engineType);
+        TriggerService triggerService = new TriggerService(Settings.EMPTY, Collections.singleton(triggerEngine));
+
+        Trigger trigger = mock(Trigger.class);
+        when(trigger.type()).thenReturn(engineType);
```

Comment:
Only a nit: It does not influence the test in any way but at first I was confused that you return `engineType` as result of the `trigger.type()` call. Copy & paste error?

Reply:
this is intentional, as otherwise the `TriggerService.addToStats()`/`TriggerService.add()` methods both would NPE, and also the reason why there is a "foo" mock engine added... hope that explains it

Reply:
I see, thanks for the explanation.